services/src/datasets/external/netcdfcf/mod.rs

```diff
@@ -0,0 +1,1109 @@
+            .context(error::MissingCrs)?;
+        let spatial_reference: SpatialReferenceOption =
+            SpatialReference::from_str(&spatial_reference)
+                .context(error::CannotParseCrs)?
+                .into();
+
+        let entities = root_group
+            .dimension_as_string_array("entities")
+            .context(error::MissingEntities)?;
+
+        let mut group_deque: VecDeque<GroupPath> = root_group
+            .group_names()
+            .into_iter()
+            .map(|name| Ok(GroupPath::new(root_group.open_group(&name)?)))
+            .collect::<Result<VecDeque<_>>>()?;
+
+        let mut datasets = Vec::new();
+
+        while let Some(group_path) = group_deque.pop_front() {
+            let subgroups = group_path
+                .current
+                .group_names()
+                .into_iter()
+                .map(|name| Ok(group_path.extend(group_path.current.open_group(&name)?)))
+                .collect::<Result<Vec<_>>>()?;
+
+            if !subgroups.is_empty() {
+                group_deque.extend(subgroups);
+
+                continue;
+            }
+
+            // Emit datasets
+
+            let (mut group_names, mut group_titles): (Vec<String>, Vec<String>) = group_path
+                .prefix
+                .into_iter()
+                .map(|s| (s.name, s.title))
+                .unzip();
+            group_names.push(group_path.current.name.clone());
+            group_titles.push(
+                group_path
+                    .current
+                    .attribute_as_string("standard_name")
+                    .unwrap_or_default(),
+            );
+
+            let group_name_path = group_names.join("/");
+            let group_title_path = group_titles.join(" > ");
+
+            let data_type = group_path.current.datatype_of_numeric_array("ebv_cube")?;
+
+            for (entity_idx, entity_name) in entities.iter().enumerate() {
+                datasets.push(DatasetListing {
+                    id: DatasetId::External(ExternalDatasetId {
+                        provider_id: id,
+                        dataset_id: format!("{file_name}::{group_name_path}::{entity_idx}"),
+                    }),
+                    name: format!("{title}: {group_title_path} > {entity_name}"),
+                    description: "".to_owned(), // TODO: where to get from file?
+                    tags: vec![],               // TODO: where to get from file?
+                    source_operator: "GdalSource".to_owned(),
+                    result_descriptor: TypedResultDescriptor::Raster(RasterResultDescriptor {
+                        data_type,
+                        spatial_reference,
+                        measurement: Measurement::Unitless, // TODO: where to get from file?
+                        no_data_value: None, // we don't want to open the dataset at this point. We should get rid of the result descriptor in the listing in general
```

Comment:
you can add a task for that. I guess we need to rework the Metadata stuff in general